/*
 * This file was automatically generated by EvoSuite
 * Thu Feb 16 02:27:36 GMT 2017
 */

package dominion;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import dominion.Main;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.Random;
import org.evosuite.runtime.TooManyResourcesException;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false) 
public class Main_ESTest extends Main_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      String[] stringArray0 = new String[7];
      stringArray0[0] = "4";
      stringArray0[1] = "WbXNw')=!QixtD=[G";
      stringArray0[2] = "";
      stringArray0[3] = "";
      stringArray0[4] = "";
      stringArray0[5] = "#*b8RzCzN*";
      stringArray0[6] = "javax.xml.datatype.Duration#isSet(DatatypeConstants.Field field) ";
      Main.main(stringArray0);
      Main.main(stringArray0);
      Main.main(stringArray0);
      Random.setNextRandom(3);
      Main.main(stringArray0);
      Main.main(stringArray0);
      // Undeclared exception!
      try { 
        Main.main(stringArray0);
        fail("Expecting exception: TooManyResourcesException");
      
      } catch(TooManyResourcesException e) {
         //
         // Loop has been executed more times than the allowed 10000
         //
         assertThrownBy("org.evosuite.runtime.LoopCounter", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      // Undeclared exception!
      try { 
        Main.main((String[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("dominion.Main", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Main main0 = new Main();
      Main main1 = new Main();
      Main main2 = new Main();
      String[] stringArray0 = new String[0];
      // Undeclared exception!
      try { 
        Main.main(stringArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 0
         //
         assertThrownBy("dominion.Main", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      String[] stringArray0 = new String[4];
      stringArray0[0] = "h6HKiP2MIVq";
      stringArray0[1] = "";
      stringArray0[2] = "fXm%cMl`womY";
      stringArray0[3] = "`[sen2[";
      // Undeclared exception!
      try { 
        Main.main(stringArray0);
        fail("Expecting exception: NumberFormatException");
      
      } catch(NumberFormatException e) {
         //
         // For input string: \"h6HKiP2MIVq\"
         //
         assertThrownBy("java.lang.NumberFormatException", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      String[] stringArray0 = new String[0];
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      Main main0 = new Main();
      Main main1 = new Main();
      Main main2 = new Main();
      Main main3 = new Main();
      Main main4 = new Main();
      Main main5 = new Main();
      Main main6 = new Main();
      Main main7 = new Main();
      Main main8 = new Main();
      Main main9 = new Main();
      Main main10 = new Main();
      Random.setNextRandom(5);
  }
}
